### DISCUSSION TITLE
How do I connect a logger in some submodule

### DISCUSSION THREAD

**Discussion by kajen3** [PrefectHQ/prefect#8065] [Q&A]
How can I use a logger inside a function imported from another module? 
Without prefect I would simple define a logger via `logger=logging.getLogger(__name__)` inside each submodule (f.e.`foo.py`).
However when I use this approach the log-messages from the `bar`-function are not displayed as can be seen on the screenshot below.
What would be the correct approach? 

```python
# log_flow.py
import sys

import prefect
from foo import bar
from prefect import flow, get_run_logger, task
from utilities import AN_IMPORTED_MESSAGE


@task
def log_task(name):
    logger = get_run_logger()
    logger.info("Hello %s!", name)
    logger.info("Prefect Version = %s 🚀", prefect.__version__)
    logger.debug(AN_IMPORTED_MESSAGE)


@flow()
def log_flow(name: str):
    logger = get_run_logger()
    logger.info("Hello")
    log_task(name)


if __name__ == "__main__":
    name = sys.argv[1]
    log_flow(name)
```

```python
# foo.py
import logging

logger = logging.getLogger("prefect.flow_runs.foo")

def bar():
    logger.info("Info")
    logger.warning("Warning")
    a = 3 + 3
    return a

```

<img width="685" alt="Screenshot 2023-01-05 at 08 32 02" src="https://user-images.githubusercontent.com/82041909/210725866-765ac842-a2f5-4449-9653-e6f661e6fa3b.png">

**Comment by zanieb** [comment]
Hi! Sounds like you're looking for the [`PREFECT_LOGGING_EXTRA_LOGGERS`](https://docs.prefect.io/api-ref/prefect/settings/#prefect.settings.PREFECT_LOGGING_EXTRA_LOGGERS) setting

**Reply by athrpf** [reply]
@zanieb Can you give an example how exactly this should work? I have tried the following, but it does not log anything into the eith er the command line or the UI:

```
# my_flow.py
import os
from prefect import task, flow
import logging

logger = logging.getLogger("foo")


@task
def log_message(message: str):
    logger.info(message)


@flow
def logged_flow():
    log_message("Hello, world!")


if __name__ == "__main__":
    os.environ["PREFECT_LOGGING_EXTRA_LOGGERS"] = "foo"
    logged_flow()
```

I have also tried starting this script with  

```
PREFECT_LOGGING_EXTRA_LOGGERS=foo python my_flow.py
```
